File: katas/java/src/test/java/mwhelan/FizzBuzzWhizzTest.java
package mwhelan;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

/**
 * @author Mike Lowe
 */
public final class FizzBuzzWhizzTest {

  @Test
  public void evaluate_1_returns1() {
    assertEquals("1", FizzBuzzWhizz.evaluate(1));
  }

  @Test
  public void evaluate_2_returnsWhizz() {
    assertEquals("Whizz", FizzBuzzWhizz.evaluate(2));
  }

  @Test
  public void evaluate_3_returnsFizzWhizz() {
    assertEquals("FizzWhizz", FizzBuzzWhizz.evaluate(3));
  }

  @Test
  public void evaluate_4_returns4() {
    assertEquals("4", FizzBuzzWhizz.evaluate(4));
  }

  @Test
  public void evaluate_5_returnsBuzzWhizz() {
    assertEquals("BuzzWhizz", FizzBuzzWhizz.evaluate(5));
  }

  @Test
  public void evaluate_6_returnsFizz() {
    assertEquals("Fizz", FizzBuzzWhizz.evaluate(6));
  }

  @Test
  public void evaluate_7_returnsWhizz() {
    assertEquals("Whizz", FizzBuzzWhizz.evaluate(7));
  }

  @Test
  public void evaluate_8_returns8() {
    assertEquals("8", FizzBuzzWhizz.evaluate(8));
  }

  @Test
  public void evaluate_9_returnsFizz() {
    assertEquals("Fizz", FizzBuzzWhizz.evaluate(9));
  }

  @Test
  public void evaluate_10_returnsBuzz() {
    assertEquals("Buzz", FizzBuzzWhizz.evaluate(10));
  }

  @Test
  public void evaluate_11_returnsWhizz() {
    assertEquals("Whizz", FizzBuzzWhizz.evaluate(11));
  }

  @Test
  public void evaluate_15_returnsFizzBuzz() {
    assertEquals("FizzBuzz", FizzBuzzWhizz.evaluate(15));
  }

  @Test
  public void evaluate_20_returnsBuzz() {
    assertEquals("Buzz", FizzBuzzWhizz.evaluate(20));
  }

  @Test
  public void evaluate_30_returnsFizzBuzz() {
    assertEquals("FizzBuzz", FizzBuzzWhizz.evaluate(30));
  }

}

File: katas/javascript/mwhelan/test/FizzBuzzWhizzTest.js
var assert = require('assert');
var FizzBuzzWhizz = require('../src/FizzBuzzWhizz.js');

describe('FizzBuzzWhizz', () => {
  describe('#run()', () => {
    it('6 should return Fizz', () => {
      assert.equal('Fizz', FizzBuzzWhizz.run(6));
    });
    it('9 should return Fizz', () => {
      assert.equal('Fizz', FizzBuzzWhizz.run(9));
    });
    it('10 should return Buzz', () => {
      assert.equal('Buzz', FizzBuzzWhizz.run(10));
    });
    it('20 should return Buzz', () => {
      assert.equal('Buzz', FizzBuzzWhizz.run(10));
    });
    it('15 should return FizzBuzz', () => {
      assert.equal('FizzBuzz', FizzBuzzWhizz.run(15));
    });
    it('30 should return FizzBuzz', () => {
      assert.equal('FizzBuzz', FizzBuzzWhizz.run(30));
    });
    it('3 should return FizzWhizz', () => {
      assert.equal('FizzWhizz', FizzBuzzWhizz.run(3));
    });
    it('5 should return BuzzWhizz', () => {
      assert.equal('BuzzWhizz', FizzBuzzWhizz.run(5));
    });
    it('1 should return 1', () => {
      assert.equal(1, FizzBuzzWhizz.run(1));
    });
    it('4 should return 4', () => {
      assert.equal(4, FizzBuzzWhizz.run(4));
    });
    it('2 should return Whizz', () => {
      assert.equal('Whizz', FizzBuzzWhizz.run(2));
    });
    it('7 should return Whizz', () => {
      assert.equal('Whizz', FizzBuzzWhizz.run(7));
    });
  });
});
